DEP Update package versions; add pyarrow and cloudpickle

stephen-hoover opened the pull request:
### New packages
- cloudpickle 0.3.1
- pyarrow 0.5.0 (from conda-forge)

### Python
- Update from v3.6.1 to v3.6.2

### Package Updates
- civis 1.5.2 -> 1.6.0
- cython 0.25.2 -> 0.26
- ipython 6.0.0 -> 6.1.0
- jinja2 2.8 -> 2.9.6
- numpy 1.12.1 -> 1.13.1
- pandas 0.20.1 -> 0.20.3
- pytest 3.0.5 -> 3.1.3
- seaborn 0.7.1 -> 0.8
- scipy 0.19.0 -> 0.19.1
- scikit-learn 0.18.1 -> 0.18.2
- pubnub 4.0.10 -> 4.0.12
- requests-toolbelt 0.7.1 -> 0.8.0
- tensorflow 1.1.0 -> 1.2.1

Commit:
DEP Update package versions; add pyarrow and cloudpickle

- Python v3.6.1 -> v3.6.2

- cloudpickle 0.3.1
- pyarrow 0.5.0 (from conda-forge)

- civis 1.5.2 -> 1.6.0
- cython 0.25.2 -> 0.26
- ipython 6.0.0 -> 6.1.0
- jinja2 2.8 -> 2.9.6
- numpy 1.12.1 -> 1.13.1
- pandas 0.20.1 -> 0.20.3
- pytest 3.0.5 -> 3.1.3
- seaborn 0.7.1 -> 0.8
- scipy 0.19.0 -> 0.19.1
- scikit-learn 0.18.1 -> 0.18.2
- pubnub 4.0.10 -> 4.0.12
- requests-toolbelt 0.7.1 -> 0.8.0
- tensorflow 1.1.0 -> 1.2.1

Also install `xgboost` from conda-forge instead of from PyPI, and add a test to verify that we aren't getting more than expected from conda-forge.

beckermr commented:
If we are pulling from conda forge, we should pull xgboost from their too. This should solve our mac packaging problems for people doing local installs of the environment.

beckermr commented:
Ack. Their should be there. English is hard.

beckermr commented on environment.yml:
```diff
@@ -17,32 +20,34 @@ dependencies:
+  - requests-toolbelt==0.8.0
+  - tensorflow==1.2.1
   - urllib3==1.19
   - xgboost==0.6a2
```
Move this above to get xgboost from conda forge.

beckermr replied:
In fact, we could probably be getting a lot of this stuff from conda forge if we want to go all in on it.

stephen-hoover commented:
I moved `xgboost` to the conda installs, and added a test to verify that we're installing the expected number of packages from conda-forge. Apologies for squashing, I wasn't thinking.

beckermr commented:
No problem on squashing. The diff is small. Nice test for conda-forge installs. This can be tricky. 

LGTM!

beckermr reviewed (approved):
LGTM!
